front/app/containers/IdeasNewSurveyPage/IdeasNewSurveyForm/index.tsx

```diff
@@ -250,6 +252,40 @@ const IdeasNewSurveyForm = ({ project, phaseId }: Props) => {
     }
   };
 
+  // GeoJSON values in the data may contain nested arrays, which Rails strong parameters cannot handle.
+  // This function converts GeoJSON values to 'well known text' (WKT) format before submitting the form(s).
+  // The BE will then convert the WKT back to GeoJSON, if valid, before saving the data.
+  function convertGeojsonToWKT(rawData: any) {
+    const data = Object.assign({}, rawData);
+
+    for (const key in data) {
+      if (typeof data[key] === 'object') {
+        for (const subKey in data[key]) {
+          if (['Point', 'LineString', 'Polygon'].includes(data[key][subKey])) {
+            const coordinates = data[key]['coordinates'].flat(2);
+            let coordinatesString = '';
+
+            for (let i = 0; i < coordinates.length; i++) {
+              coordinatesString += coordinates[i];
+              if (i < coordinates.length - 1) {
+                coordinatesString += i % 2 === 1 ? ', ' : ' ';
+              }
+            }
+
+            const nesting = data[key]['type'] === 'Polygon' ? 2 : 1;
+            data[key] =
+              `${data[key]['type'].toUpperCase()} ` +
+              `${'('.repeat(nesting)}${coordinatesString}${')'.repeat(
+                nesting
+              )}`;
+          }
+        }
+      }
+    }
+
+    return data;
+  }
```

Comment:
We can probably replace all this with a use of https://www.npmjs.com/package/wkt

Reply:
Yeah you could also use a package, but this one has zero tests and also hasn't been maintained in five years. You can also use this function, but maybe I would extract it function into a separate `utils` file and add some unit tests to it

Reply:
Moved the function to a util file & added tests :) :+1: